src/nvim/drawline.c

```diff
@@ -52,6 +55,19 @@ typedef struct {
 # include "drawline.c.generated.h"
 #endif
 
+// possible draw states in win_line()
+typedef enum {
+  WL_START = 0,  // nothing done yet
+  WL_CMDLINE,    // cmdline window column
+  WL_FOLD,       // 'foldcolumn'
+  WL_SIGN,       // column for signs
+  WL_NR,         // line number
+  WL_GR,         // 'gutter'
+  WL_BRI,        // 'breakindent'
+  WL_SBR,        // 'showbreak' or 'diff'
+  WL_LINE,       // text in the line
+} DrawState;
```

Comment:
Needed the `draw_state` defines in the `get_gutter_display_info()` function. I think I gathered from previous PRs that enums like are preferred over defines. Is this a welcome change? Could break it out into a separate PR again if preferred.